snmplib/snmp_api.c

```diff
@@ -5464,6 +5464,10 @@ snmp_free_var_internals(netsnmp_variable_list * var)
         SNMP_FREE(var->name);
     if (var->val.string != var->buf)
         SNMP_FREE(var->val.string);
+    if (var->val.bitstring)
+        SNMP_FREE(var->val.bitstring);
+    if (var->val.objid)
+        SNMP_FREE(var->val.objid);
```

Comment:
Given that string, bitstring and objid are all pointers inside the same union, can you please explain how these are different than operating on `string`?

```C
typedef union {
   long           *integer;
   u_char         *string;
   oid            *objid;
   u_char         *bitstring;
   struct counter64 *counter64;
#ifdef NETSNMP_WITH_OPAQUE_SPECIAL_TYPES
   float          *floatVal;
   double         *doubleVal;
   /*
    * t_union *unionVal; 
    */
#endif                          /* NETSNMP_WITH_OPAQUE_SPECIAL_TYPES */
} netsnmp_vardata;
```